test/extensions/filters/network/http_connection_manager/dependency_manager_test.cc

```diff
@@ -32,6 +33,39 @@ TEST(DependencyManagerTest, RegisterFilterWithDependency) {
   EXPECT_TRUE(result.ok());
 }
 
+// Register five filters. The internal filter chain construct is modified
+// iff at least one dependency is required or provided.
+TEST(DependencyManagerTest, RegisterFilterIffNonEmptyDependencies) {
+  DependencyManager manager;
+  FilterDependencies dependencies;
+  Dependency potato;
+  potato.set_type(Dependency::FILTER_STATE_KEY);
+  potato.set_name("potato");
+
+  manager.registerFilter("a", dependencies);
+  EXPECT_EQ(manager.filterChainSizeForTest(), 0);
+
+  dependencies.Clear();
+  *(dependencies.add_decode_required()) = potato;
```

Comment:
I would just recreate the dependency proto everytime instead of clearing it to make the tests a bit easier to follow

Reply:
Done.